src/cat/cat.c

```diff
@@ -1,6 +1,325 @@
+#include <ctype.h>
+#include <stdbool.h>
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#define NAME "cat (canoutils)"
+#define VERSION "1.0.0"
+#define AUTHOR "Akos Szijgyarto (SzAkos04)"
+
+#define print_version()                                                        \
+  do {                                                                         \
+    printf("%s\nversion: %s\nby: %s\n", NAME, VERSION, AUTHOR);                \
+  } while (0)
+
+#define print_help()                                                           \
+  do {                                                                         \
+    printf("Usage: cat [OPTION]... [FILE]...\n");                              \
+    printf("Concatenate FILE(s) to standard output.\n");                       \
+  } while (0)
+
+#define print_incorrect_args()                                                 \
+  do {                                                                         \
+    printf("incorrect arguments\n");                                           \
+    printf("see `cat --help`\n");                                              \
+  } while (0)
+
+#define BUF_MAX_LEN 4096 // max length of a buffer in bytes
+#define PATH_LEN 256     // max length of a path in bytes
+#define ARGS_MAX 16      // number of the max arguments
+#define ARGS_LEN 32      // max length of the arguments in bytes
+
+bool number_nonblank = false;  // number nonempty output lines, overrides -n
+bool show_ends = false;        // display $ at end of each line
+bool number = false;           // number all output lines
+bool squeeze_blank = false;    // suppress repeated empty output lines
+bool show_tabs = false;        // display TAB characters as ^I
+bool show_nonprinting = false; // use ^ and M- notation, except for LFD and TAB
+
+int cat(int filec, char **paths);
+int print_file(char *buf);
+int print_stdin(void);
+
+int main(int argc, char **argv) {
+  if (argc < 2) {
+    // print stdin
+    if (print_stdin() != 0) {
+      return 1;
+    }
+    return 0;
+  }
```

Comment:
`print_stdin` already return either `0` or `1`:
```suggestion
    return (print_stdin() == 0) ? EXIT_SUCCESS : EXIT_FAILURE;
  }
```